### ISSUE TITLE
Highlighting multiple specific lines offset by 1

### ISSUE THREAD

**Issue by grikomsn** [shikijs/twoslash#154]
Hi there, been a huge fan of twoslash!

Currently experimenting with `remark-shiki-twoslash` on my Next.js + `@mdx-js/mdx` v2 project and for some reason when highlighting multiple specific lines (e.g. `{7,12}`) **it offsets by one line**. No custom configurations and no custom stylesheets (only using [provided from readme](https://github.com/shikijs/twoslash/tree/main/packages/remark-shiki-twoslash#plugin-setup)), only using `rehype-raw` to fix #125.

Take for example, this snippet ([taken here](https://griko.id/blog/adaptive-link-element-for-nextjs#wrap-check)):

    ```jsx {7,12}
    function CustomLink(props) {
      const isRelative = props.href?.startsWith("/") ?? false;

      if (isRelative) {
        return (
          <Link href={props.href}>
            <a {...props} />
          </Link>
        );
      }

      return <a {...props} target="_blank" />;
    }
    ```

Produces this (using [`remark-shiki-twoslash`](https://github.com/shikijs/twoslash/tree/main/packages/remark-shiki-twoslash)):

<img width="1382" alt="image" src="https://user-images.githubusercontent.com/8220954/169720507-8251a076-681a-47ff-85f9-bbf8a7707a2e.png">

Which it not correct and should produce this (using [`rehype-pretty-code`](https://github.com/atomiks/rehype-pretty-code)):

<img width="1372" alt="image" src="https://user-images.githubusercontent.com/8220954/169720522-d6cb1ced-e783-492e-a92d-eb7cb5c1e928.png">

My guess it's somewhere around [this line](https://github.com/shikijs/twoslash/blob/d5510d7e589a4d53e40c8c34ade7bbe0826fdb55/packages/shiki-twoslash/src/renderers/twoslash.ts#L71) but I'll try experimenting if it's from my end of from `shiki-twoslash`'s end:

https://github.com/shikijs/twoslash/blob/d5510d7e589a4d53e40c8c34ade7bbe0826fdb55/packages/shiki-twoslash/src/renderers/twoslash.ts#L71

Anything I did wrong? I'd love to make a reproducible project for this particular case if anyone's interested.

**Comment by orta** [comment]
I might not accept a PR changing this, unless you can prove _really_ is a bug and not just that you expected rehype-pretty-code's style and got ours instead.  Because I probably spent some time thinking about whether we want to be 0 based of 1 based with the line settings and churn here is extremely expensive for everyone.

**Comment by orta** [comment]
Yep, confirming this is intentional to match editors which start with line 1 and [gatsby-remark-vscode](https://github.com/andrewbranch/gatsby-remark-vscode#line-highlighting) which we share infra with

I'm not against having a config setting like `highlightsUseZeroIndexing: true` which can configure this to help migrations across tools.

**Comment by jacob-8** [comment]
@orta - Can you please add a little more documentation at least to the shiki-twoslash readme about the line highlighting discrepancy between `twoslash` mode and not using `twoslash`. As seen from your playground, when using the `twoslash` meta attribute, highlighting is line 1 based:
![image](https://user-images.githubusercontent.com/7559478/209285527-1d0c218d-da8c-4750-abe4-0a3cea8a8820.png)

And without it is line 0 based:
![image](https://user-images.githubusercontent.com/7559478/209285494-90da6d99-1715-41ca-bbad-9b474eefe26f.png)

This is due to the above mentioned difference between the `defaultShikiRenderer`: https://github.com/shikijs/twoslash/blob/454c1b7ece7b05f669c61af4b4a535475b2adbab/packages/shiki-twoslash/src/renderers/shiki.ts#L27

and the `twoslashRenderer` https://github.com/shikijs/twoslash/blob/454c1b7ece7b05f669c61af4b4a535475b2adbab/packages/shiki-twoslash/src/renderers/twoslash.ts#L71

That was a little confusing for me. I'm going to circumvent it at the moment by just adding 1 to line numbers when the `twoslash` meta is not provided.

I love this tool by the way! I'm very close to publishing an `mdsvex-shiki-twoslash` plugin that you can use as a highlighter in MDSvex. (MDSvex sadly doesn't play nice with shiki as a remark plugin at the moment).

**Comment by orta** [comment]
Sounds more like a bug to me, they should definitely be consistent 👍🏻
